build-system/eslint-rules/no-global.js

```diff
@@ -17,12 +17,17 @@
 var astUtils = require('eslint/lib/ast-utils');
 
 var GLOBALS = Object.create(null);
-GLOBALS.window = true;
+GLOBALS.window = 'Use `self` instead.';
+GLOBALS.document = 'Reference it as `self.document` or similar instead.';
 
 module.exports = function(context) {
   return {
     Identifier: function(node) {
-      if (!GLOBALS[node.name]) {
+      var name = node.name;
+      if (!(name in GLOBALS)) {
+        return;
+      }
+      if (!(/Expression$/.test(node.parent.type))) {
```

Comment:
`/Expression$/` is a regex. I'm making sure the parent node is some `Expression` (Javascript AST stuff).